### DISCUSSION TITLE
Range sliders

### DISCUSSION THREAD

**Discussion by MrAliSalehi** [emilk/egui#3771] [Ideas]
Im not sure if the title is the correct name for this, and im not sure if it already exist, I looked for it but I didnt find anything.
take a look at this code:

```rust
ui.horizontal(|ui| {
    ui.label("from:");
    let from = self.from;
    ui.add(egui::Slider::new(&mut self.from, 0..=from + 30)
        .smart_aim(false)
        .drag_value_speed(1.0)
        .step_by(1.0));

    ui.label("to:");
    ui.add(egui::Slider::new(&mut self.to, self.from + 30..=350)
        .smart_aim(false)
        .drag_value_speed(1.0)
        .step_by(1.0));
});
```

I'm trying to get a `Range` from user, so basically a `From Number` & `To Number` , the provided code "kinda" works but its not really beautiful, its a bit smelly and poor😂.

**Comment by dumski** [comment]
Hi, I've looked some time for this feature in EGUI and I'm little surprised that there's so little mentions and requests for this feature. I think that in a such mature data-centered UI like EGUI it's a must-have. I believe it's not a very difficult to create Slider variant for a range of values - or am I missing something?

BTW, big thanks to developers for such a great job so far!

**Comment by stefnotch** [comment]
The relevant feature request for this seems to be https://github.com/emilk/egui/issues/2744
